src/Surfnet/ServiceProviderDashboard/Application/Metadata/JsonGenerator.php

```diff
@@ -67,11 +67,14 @@ public function generateForNewEntity(ManageEntity $entity, string $workflowState
         ];
     }
 
-    public function generateForExistingEntity(ManageEntity $entity, string $workflowState): array
-    {
+    public function generateForExistingEntity(
+        ManageEntity $entity,
+        string $workflowState,
+        string $updatedPart = ''
```

Comment:
Wasn't it possible to add a dedicated ACL JsonGenerator which implemented the GeneratorInterface to have better separation of concerns? Adding a parameter like this seems a code smell to me.
Please look into this if this if you see fit to change this.

Reply:
I would have no idea on how to do that.  Existing generators are for entity types (oidcng/saml/...), not for parts of entities like ACL.

My idea was to go one step at the time towards reforming the updates all-together to only send the necessary data.  Right now updates send the entire entity, rather than what's been changed.  This PR is the first one to alter that.  ACL updates now only send ACL data.  I think the way forward, which is out of scope for now, is to get rid of the `generateForExistingEntity` method on the generatorinterface.  And replace that with dedicated methods depending on what gets changed.  Exactly to separate concerns for the different update methods.

That is not possible within the budget remaining however, as that'd require a rather big overhaul if we did it all in one go.

Reply:
I agree to change the GeneratorInterface and I personally think this would be the time to change it to prevent introducing technical depth in the first place. But feel free to leave it as is for now. But would you then create a TD ticket in the backlog to secure this discussion in Pivotal. Thanks!

Reply:
Ticket added